### PR TITLE
helm_resource: allow images to be injected twice

### PR DESCRIPTION
**Pull request by nicks**
fixes https://github.com/tilt-dev/tilt-extensions/issues/528

Signed-off-by: Nick Santos <nick.santos@docker.com>

### CODE REVIEW

**Comment by nicks** [comment]
a lot of what i struggled with here was what the right "interface" for users should be.

i _think_ we want to error out if a user adds duplicate elements to `image_deps`, but i could be convinced otherwise

File: helm_resource/Tiltfile [hunk 1]
```diff
@@ -95,20 +95,16 @@ def helm_resource(
   if flags:
     apply_cmd.extend(flags)
 
+  dupe_deps = {}
   for i in range(len(image_deps)):
+    env['TILT_IMAGE_DEP_%s' % i] = image_deps[i]
```
Comment (nicksieger) [inline helm_resource/Tiltfile:100]: Is this var used or supposed to be used? I also notice that the help apply helper has a reference to `TILT_IMAGE_%s`, are they supposed to be the same?

**Review by nicksieger — APPROVED** [review]
Approach looks like a good balance of complexity and flexibility. Just one question

File: helm_resource/Tiltfile (hunk 1 above)
Comment (nicks) [reply helm_resource/Tiltfile:100]: ah, good call, i think we can remove this

**Pull request merged**